demos/common/cpp/models/src/detection_model_yolo.cpp

```diff
@@ -126,57 +134,73 @@ std::unique_ptr<ResultBase> ModelYolo3::postprocess(InferenceResult & infResult)
     const unsigned long resized_im_w, const unsigned long original_im_h,
     const unsigned long original_im_w,
     std::vector<DetectedObject>& objects) {
 
-    const int out_blob_h = static_cast<int>(blob->getTensorDesc().getDims()[2]);
-    const int out_blob_w = static_cast<int>(blob->getTensorDesc().getDims()[3]);
-    if (out_blob_h != out_blob_w) {
-        throw std::runtime_error("Invalid size of output " + output_name +
-            " It should be in NCHW layout and H should be equal to W. Current H = " + std::to_string(out_blob_h) +
-            ", current W = " + std::to_string(out_blob_h));
-    }
-
     // --------------------------- Extracting layer parameters -------------------------------------
     auto it = regions.find(output_name);
-    if(it == regions.end()) {
+    if (it == regions.end()) {
         throw std::runtime_error(std::string("Can't find output layer with name ") + output_name);
     }
     auto& region = it->second;
 
-    auto side = out_blob_h;
-    auto side_square = side * side;
+    auto& dims = blob->getTensorDesc().getDims();
+    int sideW = 0;
+    int sideH = 0;
+    unsigned long scaleH;
+    unsigned long scaleW;
+    if (isYoloV3) {
+        const int out_blob_h = static_cast<int>(dims[dims.size() - 2]);
+        const int out_blob_w = static_cast<int>(dims[dims.size() - 1]);
+        if (out_blob_h != out_blob_w) {
+            throw std::runtime_error("Invalid size of output " + output_name +
+                " It should be in NCHW layout and H should be equal to W. Current H = " + std::to_string(out_blob_h) +
+                ", current W = " + std::to_string(out_blob_h));
+        }
+        sideH = out_blob_h;
+        sideW = out_blob_w;
+        scaleW = resized_im_w;
+        scaleH = resized_im_h;
+    }
+    else {
+        sideH = region.outputHeight;
+        sideW = region.outputWidth;
+        scaleW = region.outputWidth;
+        scaleH = region.outputHeight;
+    }
+
+    auto entriesNum = sideW * sideH;
     const float* output_blob = blob->buffer().as<PrecisionTrait<Precision::FP32>::value_type*>();
 
     // --------------------------- Parsing YOLO Region output -------------------------------------
-    for (int i = 0; i < side_square; ++i) {
-        int row = i / side;
-        int col = i % side;
+    for (int i = 0; i < entriesNum; ++i) {
+        int row = i / sideW;
+        int col = i % sideW;
         for (int n = 0; n < region.num; ++n) {
             //--- Getting region data from blob
-            int obj_index = calculateEntryIndex(side, region.coords, region.classes, n * side * side + i, region.coords);
-            int box_index = calculateEntryIndex(side, region.coords, region.classes, n * side * side + i, 0);
+            int obj_index = calculateEntryIndex(entriesNum, region.coords, region.classes, n * entriesNum + i, region.coords);
+            int box_index = calculateEntryIndex(entriesNum, region.coords, region.classes, n * entriesNum + i, 0);
```

Comment:
Since you are here, what did you think about deleting the `obj_index` and `class_index` and replacing their usage with calculations:
```
float scale = output_blob[box_index + 4 * entriesNum];
...
for (int j = 0; j < region.classes; ++j) {
     float prob = scale * output_blob[box_index + (5+i) * entriesNum];

Reply:
If it ain't broke - don't fix it.
It's OK with me to do such refactoring, but let's do it in separate PR to avoid breaking anything here.